agent/config/config.go

```diff
@@ -188,6 +188,7 @@ type Config struct {
 	AdvertiseAddrWAN                 *string                  `json:"advertise_addr_wan,omitempty" hcl:"advertise_addr_wan" mapstructure:"advertise_addr_wan"`
 	AdvertiseAddrWANIPv4             *string                  `json:"advertise_addr_wan_ipv4,omitempty" hcl:"advertise_addr_wan_ipv4" mapstructure:"advertise_addr_wan_ipv4"`
 	AdvertiseAddrWANIPv6             *string                  `json:"advertise_addr_wan_ipv6,omitempty" hcl:"advertise_addr_wan_ipv6" mapstructure:"advertise_addr_ipv6"`
+	Audit                            Audit                    `json:"audit,omitempty"     hcl:"audit"     mapstructure:"audit"`
```

Comment:
Similar to my previous comment, I think we could create an `EnterpriseConfig` type and embed it in this struct for top level enterprise only config. Similar to how we added enterprise specific DNS configurations for namespaces:

https://github.com/hashicorp/consul/blob/e9e8c0e730ebe546d9aba16d67d7dee92129fb82/agent/config/config.go#L620

Reply:
agreed!